example-apps/todo-mvc-example/source/components/todo-list.ts

```diff
@@ -8,31 +11,50 @@ import {TodoItem} from './todo-item';
 @View({
   directives: [NgIf, NgFor, TodoItem],
   template: `
-		<section class="todoapp">
+    <section class="todoapp">
       <header class="header">
-				<h1>todos</h1>
-				<input class="new-todo" placeholder="What needs to be done?" autofocus=""
-        #newtodo (keyup)="addTodo($event, newtodo)">
-			</header>
-			<section class="main" *ng-if="todoStore.todos.length > 0">
-				<ul class="todo-list">
-					<li *ng-for="#todo of todoStore.todos" [class.completed]="todo.completed" [class.editing]="todo.editing">
-             <todo-item [todo]="todo"></todo-item>
+        <h1>todos</h1>
+        <input class="new-todo"
+          placeholder="What needs to be done?"
+          autofocus=""
+          (keyup)="addTodo($event)">
+      </header>
+      <section class="main" *ngIf="todoStore.todos.length > 0">
+        <ul class="todo-list">
+          <li *ngFor="#todo of todos"
+            [class.completed]="todo.completed"
+            [class.editing]="todo.editing">
+              <todo-item [todo]="todo"></todo-item>
           </li>
-				</ul>
-			</section>
-		</section>`
+        </ul>
+      </section>
+    </section>`
 })
-export class TodoList {
+export class TodoList implements OnInit, OnDestroy {
+  public todos: Todo[] = [];
+  private _subscription: Subscription<Todo[]>;
 
   constructor(@Inject(TodoStore) private todoStore: TodoStore) {
-    this.todoStore = todoStore;
   }
 
-  addTodo($event, newtodo) {
-    if ($event.which === 13 && newtodo.value.trim().length) {
-      this.todoStore.add(newtodo.value);
-      newtodo.value = '';
+  public addTodo($event) {
```

Comment:
I like to omit public for brevity as it is the default, but do not have a strong preference